src/main/scala/fs2/aws/kinesis/KinesisWorkerSettings.scala

```diff
@@ -0,0 +1,31 @@
+package fs2
+package aws
+package kinesis
+
+import scala.concurrent.duration._
+
+case class KinesisWorkerStreamSettings(
+  bufferSize: Int,
+  backpressureTimeout: FiniteDuration) {
+  require(
```

Comment:
is that good practice ? Should you not have a class and a companion object with a constructor that returns an Either instead?

Reply:
good question! I'm not sure on what the best practice is for scala around the use of `require` and `assert`. Either way, I think its more functional to return Either, so I'll go down that route